### PR TITLE
Add conversions to Linkedin Ads Source package

### PR DESCRIPTION
**Pull request by fivetran-avinash**
## PR Overview
**This PR will address the following Issue/Feature:** Adding conversion support

**This PR will result in the following new package version:** 0.9.0

Although I don't think there are no breaking changes per se, there is enough new functionality added with conversions that this warrants a version upgrade.

**Please provide the finalized CHANGELOG entry which details the relevant changes included in this PR:**

## Feature Updates: Conversion Support!
We have added more robust support for conversions in our data models by doing the following: 
- Created a `linkedin_ads__conversion_fields` variable to pass through additional conversion metrics in the `stg_linkedin_ads__ad_analytics_by_campaign` and `stg_linkedin_ads__ad_analytics_by_creative` models. 
  - Set variable defaults in the `dbt_project.yml` to bring in the most used conversion fields `external_website_conversions` and `one_click_leads`. 
- Ensured backwards compatibility with existing passthrough column variables in these models by creating macro checks for whether these fields already are brought in by the existing passthrough variables. This ensures there are no duplicate column errors if both the new conversion variable and the old passthrough variable are leveraged in either `stg_linkedin_ads__ad_analytics` data model. 
- Brought in the `conversion_value_in_local_currency` field to the above mentioned `stg_linkedin_ads__ad_analytics` models.

## Documentation Update 
- Documents how to set your own passthrough fields with the variable `linkedin_ads__conversion_fields` [in the README](https://github.com/fivetran/dbt_linkedin_source/blob/main/README.md#adding-in-conversion-fields-variable).

## Under the Hood
- Updated `linkedin_ad_analytics_by_creative_data` seed file with relevant conversion fields for more robust testing. 

### Basic Validation
Please acknowledge that you have successfully performed the following commands locally:
- [x] dbt run –full-refresh && dbt test
- [ ] dbt run (if incremental models are present) && dbt test

Before marking this PR as "ready for review" the following have been applied:
- [NA] The appropriate issue has been linked, tagged, and properly assigned.
- [x] All necessary documentation and version upgrades have been applied.
    
- [x] docs were regenerated (unless this PR does not include any code or yml updates).
- [x] BuildKite integration tests are passing.
- [ ] Detailed validation steps have been provided below.

### Detailed Validation
Please share any and all of your validation steps:

See transform PR. 

### If you had to summarize this PR in an emoji, which would it be?
 
🥼

### CODE REVIEW

File: CHANGELOG.md
Comment (fivetran-jamie) [inline CHANGELOG.md:19]:
```suggestion
- [Seer Interactive](https://www.seerinteractive.com/?utm_campaign=Fivetran%20%7C%20Models&utm_source=Fivetran&utm_medium=Fivetran%20Documentation)
```

**Review by fivetran-jamie — COMMENTED** [review]
looks great!! just doc questions/suggestions really

File: macros/linkedin_ads_add_pass_through_columns.sql
Comment (fivetran-jamie) [inline macros/linkedin_ads_add_pass_through_columns.sql:28]:
```suggestion
```

Comment (fivetran-jamie) [reply macros/linkedin_ads_add_pass_through_columns.sql:28]: all commented out

File: README.md
Comment (fivetran-jamie) [inline README.md:43]: can you add the Ad Reporting link to line 44 as well?

File: README.md
Comment (fivetran-jamie) [inline README.md:88]:
```suggestion
By default, this package will select `clicks`, `impressions`, `cost` and `conversion_value_in_local_currency` (as well as fields set via `linkedin_ads__conversion_fields` in the next section) from the source reporting tables `ad_analytics_by_campaign` and `ad_analytics_by_creative` to store into the corresponding staging models. If you would like to pass through additional metrics to the staging models, add the below configurations to your `dbt_project.yml` file. These variables allow for the pass-through fields to be aliased (`alias`) and transformed (`transform_sql`) if desired, but not required. Only the `name` of each metric field is required. Use the below format for declaring the respective pass-through variables:
```

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md:21]: Were there breaking changes? No new models were added, just fields.

File: macros/linkedin_ads_add_pass_through_columns.sql
Comment (fivetran-avinash) [reply macros/linkedin_ads_add_pass_through_columns.sql:28]: We should keep line 28 though right?

File: README.md
Comment (fivetran-avinash) [reply README.md]: Updated.

File: README.md
Comment (fivetran-avinash) [reply README.md:43]: Added.

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md]: Updated.

File: README.md
Comment (fivetran-avinash) [reply README.md:122]: Added.

File: models/stg_linkedin.yml
Comment (fivetran-avinash) [reply models/stg_linkedin.yml:77]: Added documentation.

File: models/stg_linkedin.yml
Comment (fivetran-avinash) [reply models/stg_linkedin.yml:38]: Added documentation.

File: models/src_linkedin.yml
Comment (fivetran-avinash) [reply models/src_linkedin.yml]: Added documentation.

File: models/src_linkedin.yml
Comment (fivetran-avinash) [reply models/src_linkedin.yml:42]: Added documentation.

File: README.md
Comment (fivetran-avinash) [reply README.md]: Updated.

File: README.md
Comment (fivetran-avinash) [reply README.md]: Updated.

File: README.md
Comment (fivetran-avinash) [reply README.md]: Added documentation.

File: README.md
Comment (fivetran-avinash) [reply README.md]: Updates applied.

File: DECISIONLOG.md
Comment (fivetran-avinash) [reply DECISIONLOG.md]: Updated.

File: DECISIONLOG.md
Comment (fivetran-avinash) [reply DECISIONLOG.md]: Updated.

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md:23]: Agreed.

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md]: Added it 😅

File: CHANGELOG.md
Comment (fivetran-avinash) [reply CHANGELOG.md]: Added

**Comment by fivetran-avinash** [comment]
> lgtm! one minor comment in changelog but that's truly it

Fixed!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
@fivetran-jamie this PR looks great and will really add a lot of value to our LinkedIn data models! 🎉 

This PR generally looks great but I do have one small callout in the README. I was stress testing the variables and only ran into an issue when I tried to pass the `alias` of a passthrough column to the conversions variable. I don't imagine we need to update the code to account for this, but we should make this clear in the README. Would you be able to update the wording and let me know once it's updated. Once I approve this PR it would be great to apply the same documentation update to the other PR (LinkedIn transform).

That by no means has to halt the approval of this PR, but please be sure to make this update and this should be good to go!

File: README.md
Comment (fivetran-joemarkiewicz) [inline README.md:118]: I did some tests and found that this variable **NEEDS** to be the name of the field as it originally is from the source. We should clarify to customers that the field input into this variable will be the `name` and not the `alias` if there is one from the above passthrough fields.

**Pull request merged**
